### PR TITLE
Add auth.isUser to vaults endpoints + nits

### PR DESCRIPTION
**Pull request by spolu**
## Description

All our endpoints should always start by checking that `auth.isUser()` + we can use getNonNullable function since owner, plan, user have already been checked by the internal API endpoints wrappers.

## Risk

N/A

## Deploy Plan

- deploy `front`

### CODE REVIEW

File: front/pages/api/w/[wId]/labs/transcripts/[tId].ts [hunk 1]
```diff
@@ -33,18 +33,17 @@ async function handler(
   >,
   auth: Authenticator
 ): Promise<void> {
-  const userId = auth.user()?.id;
-
-  if (!userId) {
+  if (!auth.isUser()) {
```
Comment (flvndvd) [inline front/pages/api/w/[wId]/labs/transcripts/[tId].ts:36]: Are we sure that `auth.isUser()` checks for a valid plan of the workspace?

**Review by flvndvd — APPROVED** [review]
LGTM, I'm convinced we need a higher order mixin that enforces on all endpoints except for the few exceptions of the Dust apps.

File: front/pages/api/w/[wId]/labs/transcripts/[tId].ts (hunk 1 above)
Comment (spolu) [reply front/pages/api/w/[wId]/labs/transcripts/[tId].ts:36]: It does not but the wrapper does

**Comment by spolu** [comment]
> LGTM, I'm convinced we need a higher order mixin that enforces on all endpoints except for the few exceptions of the Dust apps.

Definitely agreed. But until then we want this defense in depth everywhere.

**Pull request merged**